subsys/profiler/profiler_common_shell.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <string.h>
+#include <stdlib.h>
+#include <shell/shell.h>
+#include <shell/shell_rtt.h>
+#include <profiler.h>
+
+u32_t profiler_enabled_events;
+
+static int display_registered_events(const struct shell *shell, size_t argc,
+				char **argv)
+{
+	/* Check if flags for enabling/disabling custom event fit on u32_t */
+	static_assert(CONFIG_MAX_NUMBER_OF_CUSTOM_EVENTS <=
+		      sizeof(profiler_enabled_events) * 8,
+		      "Max 32 profiler events may be used");
+	shell_fprintf(shell, SHELL_NORMAL, "EVENTS REGISTERED IN PROFILER:\n");
+	for (size_t i = 0; i < profiler_num_events; i++) {
+		const char *event_name = profiler_get_event_descr(i);
+		/* Looking for event name delimiter (',') */
+		char *event_name_end = strchr(event_name, ',');
+
+		shell_fprintf(shell,
+			      SHELL_NORMAL,
+			      "%c %d:\t%.*s\n",
+			      (profiler_enabled_events & BIT(i)) ? 'E' : 'D',
+			      i,
+			      event_name_end - event_name,
+			      event_name);
+	}
+
+	return 0;
+}
+
+static void set_event_profiling(const struct shell *shell, size_t argc,
+				char **argv, bool enable)
+{
+	u32_t evt_mask = 0;
+
+	/* If no IDs specified, all registered events are affected */
+	if (argc == 1) {
+		for (int i = 0; i < profiler_num_events; i++) {
+			evt_mask |= BIT(i);
+		}
+
+		shell_fprintf(shell,
+			      SHELL_NORMAL,
+			      "Profiling all events %s\n",
```

Comment:
`"Profiling all events %sabled\n", enable ? "en" : "dis"`

I would change it not only to save flash, but also to keep convention from other modules.

Reply:
done